streams/src/test/java/org/apache/kafka/streams/integration/RocksDBMetricsIntegrationTest.java

```diff
@@ -214,70 +176,55 @@ private StreamsBuilder builderForSegmentedStateStore() {
         return builder;
     }
 
     private void cleanUpStateRunVerifyAndClose(final StreamsBuilder builder,
                                                final Properties streamsConfiguration,
-                                               final Class outputKeyDeserializer,
-                                               final Class outputValueDeserializer,
-                                               final MetricsVerifier metricsVerifier,
-                                               final String metricsScope) throws Exception {
+                                               final MetricsVerifier metricsVerifier) throws Exception {
         final KafkaStreams kafkaStreams = new KafkaStreams(builder.build(), streamsConfiguration);
         kafkaStreams.cleanUp();
         produceRecords();
 
         StreamsTestUtils.startKafkaStreamsAndWaitForRunningState(kafkaStreams, TIMEOUT);
 
-        IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(
-            TestUtils.consumerConfig(
-                CLUSTER.bootstrapServers(),
-                "consumerApp",
-                outputKeyDeserializer,
-                outputValueDeserializer,
-                new Properties()
-            ),
-            STREAM_OUTPUT,
-            1
-        );
-        metricsVerifier.verify(kafkaStreams, metricsScope);
+        metricsVerifier.verify(kafkaStreams, "rocksdb-state-id");
+        metricsVerifier.verify(kafkaStreams, "rocksdb-window-state-id");
         kafkaStreams.close();
     }
 
     private void produceRecords() throws Exception {
         final MockTime mockTime = new MockTime(WINDOW_SIZE.toMillis());
+        final Properties prop = TestUtils.producerConfig(
+            CLUSTER.bootstrapServers(),
+            IntegerSerializer.class,
+            StringSerializer.class,
+            new Properties()
+        );
+        // non-segmented store do not need records with different timestamps
         IntegrationTestUtils.produceKeyValuesSynchronouslyWithTimestamp(
-            STREAM_INPUT,
-            Collections.singletonList(new KeyValue<>(1, "A")),
-            TestUtils.producerConfig(
-                CLUSTER.bootstrapServers(),
-                IntegerSerializer.class,
-                StringSerializer.class,
-                new Properties()
-            ),
+            STREAM_INPUT_ONE,
+            Utils.mkSet(new KeyValue<>(1, "A"), new KeyValue<>(1, "B"), new KeyValue<>(1, "C")),
+            prop,
             mockTime.milliseconds()
         );
         IntegrationTestUtils.produceKeyValuesSynchronouslyWithTimestamp(
-            STREAM_INPUT,
-            Collections.singletonList(new KeyValue<>(1, "B")),
-            TestUtils.producerConfig(
-                CLUSTER.bootstrapServers(),
-                IntegerSerializer.class,
-                StringSerializer.class,
-                new Properties()
-            ),
+            STREAM_INPUT_TWO,
+            Collections.singleton(new KeyValue<>(1, "A")),
```

Comment:
Why do we write into `STREAM_INPUT_TWO` with 3 calls instead of just one call passing in all 3 records at once?

Reply:
Note we use a mocktime with auto tick `WINDOW_SIZE.toMillis()`: and each time its `milliseconds()` is called it would auto-advance, we need the produced records with those advanced timestamps.